### PR TITLE
[FLINK-20053][table][doc] Add document for file compaction

### PR DESCRIPTION
**Pull request by JingsongLi**
## What is the purpose of the change

Add document for file compaction

## Brief change log

- In Flink FileSystem document, Add File Compaction

## Page

image

### CODE REVIEW

File: docs/dev/table/connectors/filesystem.md [hunk 1]
```diff
@@ -150,6 +150,41 @@ become finished on the next checkpoint) control the size and number of these par
 **NOTE:** For row formats (csv, json), you can set the parameter `sink.rolling-policy.file-size` or `sink.rolling-policy.rollover-interval` in the connector properties and parameter `execution.checkpointing.interval` in flink-conf.yaml together
 if you don't want to wait a long period before observe the data exists in file system. For other formats (avro, orc), you can just set parameter `execution.checkpointing.interval` in flink-conf.yaml.
 
+### File Compaction
+
+If you want a smaller checkpoint interval and do not want to generate a large number of small files,
+it is recommended that you open file compaction:
```
Comment (sjwiesman) [inline docs/dev/table/connectors/filesystem.md:156]:
```suggestion
The file sink supports file compactions, which allows applications to have smaller checkpoint intervals without generating a large number of files.
```

File: docs/dev/table/connectors/filesystem.md [hunk 2]
```diff
@@ -150,6 +150,41 @@ become finished on the next checkpoint) control the size and number of these par
...
+        <td>MemorySize</td>
+        <td>The compaction target file size, the default value is the rolling file size.</td>
+    </tr>
+  </tbody>
+</table>
+
+After you open file compaction, small files that are not large enough will be merged into large files,
+It is worth noting that:
+- Only files in a single checkpoint are compacted, that is, at least the same number of files as the number of checkpoints is generated.
+- The file before merging is invisible, so the visibility of the file may be: checkpoint interval + compaction time.
```
Comment (sjwiesman) [inline docs/dev/table/connectors/filesystem.md:186]:
```suggestion
If enabled, file compaction will merge multiple small files into larger files based on the target file size.
When running file compaction in production, please be aware that:
- Only files in a single checkpoint are compacted, that is, at least the same number of files as the number of checkpoints is generated.
- The file before merging is invisible, so the visibility of the file may be: checkpoint interval + compaction time.
```

**Review by sjwiesman — COMMENTED** [review]
Cool feature, a few comments.

**Comment by sjwiesman** [comment]
@JingsongLi +1 on the docs change but I think you accidentally committed an additional file. Feel free to merge after removing that file.

**Comment by JingsongLi** [comment]
> @JingsongLi +1 on the docs change but I think you accidentally committed two additional files. Feel free to merge after removing those.

Thanks @sjwiesman for the review, you are right, I will delete them.

**Pull request merged**